### ISSUE TITLE
Passing a positional path argument (e.g. `.`) silently suppresses all lint errors

### ISSUE THREAD

**Issue by Attempt3035** [loeffel-io/ls-lint#365]
When a positional file/dir argument is passed to `ls-lint`, violations are no longer 
reported and the process always exits `0` — even though `--debug` shows the exact 
same files being indexed and linted either way. Omitting the positional argument 
(and just running `ls-lint` from the target directory) reports errors correctly.

## Environment
- ls-lint v2.3.1 (go1.24.3)
- Reproduced on both darwin-arm64 (native binary) and Linux (via @ls-lint/ls-lint npm package)

## Repro steps
.ls-lint.yml:
​```yaml
ls:
  .abc: lowercase
​```

Directory contents:
​```
TRYSomeTHing.abc
​```

Run with no positional arg (correct):
​```
$ ls-lint --debug
...
TRYSomeTHing.abc failed for `.abc` rules: lowercase
exit: 1
​```

Run with `.` as positional arg (bug):
​```
$ ls-lint --debug .
...
exit: 0
​```

The `--debug` output (ls index, ignore index, files linted, statistics) is byte-for-byte 
identical between the two runs — the only difference is the trailing `.`. No errors are 
printed to stdout or stderr, and no error section appears anywhere in debug output.

## Expected behavior
`ls-lint .` should behave identically to `ls-lint` with no positional argument, and 
should report the `lowercase` rule violation with a non-zero exit code.

## Actual behavior
All lint errors are silently swallowed whenever any positional path argument is passed, 
regardless of the rule type (also confirmed with `screamingsnakecase` and `kebabcase` 
on separate configs/files) or whether `--warn` is used.

## Impact
This is a significant footgun since most CI setups and the CLI docs' own usage pattern 
(`ls-lint [options] [file|dir]*`) suggest passing a target path is normal usage — but 
doing so causes the linter to silently pass regardless of actual violations.
